test/Microsoft.Azure.ServiceBus.UnitTests/MessageInterop/MessageInteropEnd2EndTests.cs

```diff
@@ -0,0 +1,109 @@
+{
+    using Microsoft.Azure.ServiceBus.InteropExtensions;
+    using Microsoft.ServiceBus.Messaging;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Xunit;
+    using System;
+
+    public class MessageInteropEnd2EndTests
+    {
+        public static IEnumerable<object> TestEnd2EndEntityPermutations => new object[]
+        {
+            new object[] { TransportType.NetMessaging },
+            new object[] { TransportType.Amqp }
+        };
+
+        [Theory]
+        [MemberData(nameof(TestEnd2EndEntityPermutations))]
+        [DisplayTestMethodName]
+        async Task RunEnd2EndSerializerTests(TransportType transportType)
+        {
+            string queueName = TestConstants.NonPartitionedQueueName;
+
+            // Override and Create a new ConnectionString with SbmpConnection Endpoint scheme
+            string[] temp = TestUtility.NamespaceConnectionString.Split(':');
+            string sbConnectionString = "Endpoint=sb:" + temp[1];
+
+            if(transportType == TransportType.Amqp)
+            {
+                sbConnectionString += ';' + nameof(TransportType) + '=' + TransportType.Amqp.ToString();
+            }
+
+            // Create a full framework MessageSender
+            MessagingFactory messagingFactory = MessagingFactory.CreateFromConnectionString(sbConnectionString);
+            MessageSender fullFrameWorkClientSender = messagingFactory.CreateMessageSender(queueName);
+
+            // Create a .NetCore MessageReceiver
+            Core.MessageReceiver dotNetStandardMessageReceiver = new Core.MessageReceiver(TestUtility.NamespaceConnectionString, queueName, ServiceBus.ReceiveMode.ReceiveAndDelete);
+
+            try
+            {
+                // Plain string
+                string message1Body = "contosoString";
+                var message1 = new BrokeredMessage(message1Body);
+                await fullFrameWorkClientSender.SendAsync(message1);
+
+                // Custom object
+                var book = new TestBook("contosoBook", 1, 5);
+                var message2 = new BrokeredMessage(book);
+                await fullFrameWorkClientSender.SendAsync(message2);
+
+                // UTF8 encoded byte array object
+                string message3Body = "contosoBytes";
+                var message3 = new BrokeredMessage(Encoding.UTF8.GetBytes(message3Body));
+                await fullFrameWorkClientSender.SendAsync(message3);
+
+                // Stream Object
+                string message4Body = "contosoStreamObject";
+                var message4 = new BrokeredMessage(new MemoryStream(Encoding.UTF8.GetBytes(message4Body)));
+                await fullFrameWorkClientSender.SendAsync(message4);
+
+                // Now Receive using this clients MessageReceiver
+                // Plain string
+                var returnedMessage = await dotNetStandardMessageReceiver.ReceiveAsync();
```

Comment:
Should receiving be done by sequence number? Otherwise test implies order. highly unlikely that messages will come out of order, but still. Alternatively, send/receive pairs? Also will allow to keep the mental scope smaller.